process_creationg_termination.c

```diff
@@ -16,19 +15,17 @@ int main()
     childPid = fork();
     if(childPid  >= 0)
     {
-        if(childPid == 0)
-        {
-            printf("Child process\n");
-            sleep(5);
+        if(childPid == 0) {
+            printf("Child greeting\n");
+            sleep(2);
             printf("Child exiting\n");
         }
-        else
-        {
-            waitpid(-1,NULL,0);
+        else{
+           sleep(2);
```

Comment:
we cant just assume it to sleep for 2 seconds, if child doesnt finish in 2 seconds parent will end and child will also terminate